src/main/java/com/google/devtools/build/lib/remote/RemoteCache.java

```diff
@@ -260,6 +262,36 @@ private static Path toTmpDownloadPath(Path actualPath) {
     return actualPath.getParentDirectory().getRelative(actualPath.getBaseName() + ".tmp");
   }
 
+  /**
+   * Exception which represents a collection of IOExceptions for the purpose
+   * of distinguishing remote communication exceptions from those which occur
+   * on filesystems locally. This exception serves as a trace point for the actual
+   * download, so that the intented operation can be observed in a stack, with all
+   * constituent exceptions available for observation.
+   */
+  static class DownloadException extends Exception {
```

Comment:
It looks like this is essentially a public API. I think it would be better to move it to a top-level .java file.

Reply:
Why not make this an IOException?

Reply:
Nominally to expose this checked exception uniquely for handlers - there are too many IOException interceptors already, and I wanted to have a clear shot to the top.